[WEB-3915] -  new footer

aralovelace opened the pull request:
## Jira Ticket Link / Motivation
 
WEB-3915 - New Footer design

This is a rebase from the new Meganav [WEB-3914-new_meganav](https://github.com/ably/ably-ui/tree/WEB-3914-new_meganav) once this is approved this will be rebased in the meganav repo so they will be released together

## Summary of changes

Save the current Footer as `LegacyFooter` and replace it with the newly designed Footer. 

This pull request includes several changes to the `@ably/ui` project, focusing on updating the footer components, styles, and related data. The most important changes include the migration from `Footer` to `LegacyFooter`, updates to the footer stories, and the addition of new footer data.

Updates to footer components and stories:

* [`.storybook/styles.css`](diffhunk://#diff-fea3edddbfc8c8286686f5e8273cf20c4b1a3814b78ecf13574fdbf67ce42ecfL11-R11): Updated the import from `Footer` to `LegacyFooter` to reflect the migration.
* [`src/core/Footer/Footer.stories.tsx`](diffhunk://#diff-6eb1f879ef08273f38d52f62c434900ecf8a1af3bb05314d35fef09c6938a51eL1-R12): Removed old imports and handlers, and restructured the story to use the new `Meta` and `StoryFn` from `@storybook/react`.
* [`src/core/LegacyFooter/Footer.stories.tsx`](diffhunk://#diff-4a74fb99a2d8abf59f253f8415074b064ffa0b72415b329446e5944e25a174fcR1-R44): Added a new story for `LegacyFooter` with relevant imports and handlers.

Addition of new footer data:

* [`src/core/Footer/data.ts`](diffhunk://#diff-7b0db6bd265d641f7850bd939b65e812171468002f73db0630afbe808909ae47R1-R121): Added new data for footer links, bottom footer links, social links, and Ably awards.



## How do you manually test this?
New Footer story in the storybook
or Voltaire preview - https://voltaire-pr-966.herokuapp.com/
and in website - https://website-web-4266-apply--mvmq8w.herokuapp.com/sign-up

## Reviewer Tasks (optional)

<!-- If there is something specific you need reviewers to have a look at, something that might not be immediately clear, use this section to guide them along. -->

## Merge/Deploy Checklist

<!-- Committer checklist  -->

- [ ] Written automated tests for implemented features/fixed bugs
- [ ] Rebased and squashed commits
- [ ] Commits have clear descriptions of their changes
- [ ] Checked for any performance regressions

## Frontend Checklist

<!-- Committer frontend changes checklist  -->

- [ ] No frontend changes in this PR
- [ ] Added before/after screenshots for changes
- [ ] Tested on different platforms/browsers with [Browserstack](https://www.browserstack.com)
- [ ] Compared with the initial design / our [brand guidelines](https://www.figma.com/file/jTQgyIovrhGBOf4Zrvjvbk/Ably-Linear-Design?node-id=118%3A0)
- [ ] Checked the code for accessibility issues ([VoiceOver User Guide](https://support.apple.com/en-gb/guide/voiceover/welcome/mac))?

aralovelace commented:
@CodeRabbit review

jamiehenson reviewed (changes requested):
Looks great, nicely done! Good to see some nice flex usage in there, the learning has paid off. I've only got a couple of small comments that are quick to go through

aralovelace commented on src/core/Footer.tsx:
```diff
@@ -1,530 +1,127 @@
   return (
-    <footer
-      className="bg-light-grey font-sans antialiased leading-normal"
+    <div
```
I added the div after footer so I can add `max-w-screen-xl mx-auto ` and it wraps with w-full with its theme background color

aralovelace commented:
Hi @jamiehenson, thanks for the review! It’s now updated. For the footer, I moved the `div` inside the footer element to center the content in it,  it while keeping the background color full width.

jamiehenson reviewed (approved):
🦶 

Just remember to remove that dev package commit
